SpatialGDK/Source/SpatialGDK/Public/Interop/RPCs/RPC_RingBufferWithACK_Receiver.h

```diff
@@ -102,15 +102,15 @@ class MonotonicRingBufferWithACKReceiver : public TRPCBufferReceiver<PayloadType
 		for (auto Iterator = this->ReceivedRPCs.CreateIterator(); Iterator; ++Iterator)
 		{
 			Worker_EntityId EntityId = Iterator->Key;
-			if (!CanExtract(EntityId))
-			{
-				continue;
-			}
 			ReceiverState& State = ReceiverStates.FindChecked(EntityId);
 			auto& Payloads = Iterator->Value;
 			uint32 ProcessedRPCs = 0;
 			for (const auto& Payload : Payloads)
 			{
+				if (!CanExtract(EntityId))
```

Comment:
do we know how the truth value of this call can change while in this loop?

Reply:
Yes, for example if processing one of the Payloads triggers some logic that deletes the Actor, the truth value of the call might change. 

For example, in my debugging I have found the case in which there are 2 ServerMovePacked RPCs in the Payloads of a Pawn in which after the first ServerMovePacked RPC is applied, a OnComponentBeginOverlap callback with another Actor is called, which then results in the deletion of the Pawn. When trying to execute the second Payload (containing the second ServerMovePacked RPC), the Pawn is now deleted from the map, which causes a nullptr error.